vortex-expr/src/lib.rs

```diff
@@ -212,4 +221,213 @@ mod tests {
             "{dog:32_u32,cat:rufus}"
         );
     }
+
+    #[test]
+    fn expr_dtype() {
```

Comment:
I don't really like this test here. Aren't you testing the implementation of specific expression implementations?

Reply:
Fair point. This test is mostly a copy of expr_display above. I could fray both tests into the relevant expressions.

I don't have a strong opinion. It also seems nice to test all the variants of the same method side-by-side.

Reply:
I'll fray expr_dtype here and then PR a fraying of expr_display too.

Reply:
Done.